submissions/pozzitive11/js-world-oop/index.js

```diff
@@ -0,0 +1,65 @@
+class Inhabitant {
+  constructor(species, name, gender, saying) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.saying = saying;
+  }
+
+  getProperties() {
+    return `${this.species}; My name ${this.name}; ${this.gender}; ${this.saying}`;
+  }
+}
+class Animal extends Inhabitant {
+  constructor(name, gender, saying) {
+    super("animal", name, gender, saying);
+    this.paw = 4;
+  }
+  getProperties() {
+    return super.getProperties() + `; I have ${this.paw} paws;`;
+  }
+}
+
+class Human extends Inhabitant {
+  constructor(name, gender, saying) {
+    super("human", name, gender, saying);
+    this.hands = 2;
+    this.legs = 2;
+  }
+
+  getProperties() {
+    return (
+      super.getProperties() +
+      `; I have ${this.legs} legs and ${this.hands} hands;`
+    );
+  }
+}
+
+class Dog extends Animal {
+  constructor(name, gender) {
+    super(name, gender, "Woof-woof!");
+  }
+}
+
+class Cat extends Animal {
+  constructor(name, gender) {
+    super(name, gender, "meow-meow");
+  }
+}
+class Catwoman extends Cat {
+  constructor(name) {
+    super(name);
```

Comment:
`Cat`'s constructor called via `super` accepts `name` and `gender`. We should use this instead of trying to reassign superclass properties directly via `this.gender` as this breaks one of the SOLID principles.

Same relates to `this.species`. But hierarchical nature of classes doesn't allow to skip class levels. In similar cases consider composition pattern. You do not have to fix `this.species` handling.